sklearn/utils/mocking.py

```diff
@@ -45,19 +45,30 @@ class CheckingClassifier(BaseEstimator, ClassifierMixin):
     changed the input.
     """
     def __init__(self, check_y=None,
-                 check_X=None, foo_param=0):
+                 check_X=None, foo_param=0,
+                 expected_fit_params=None):
         self.check_y = check_y
         self.check_X = check_X
         self.foo_param = foo_param
+        self.expected_fit_params = expected_fit_params
 
-    def fit(self, X, y):
+    def fit(self, X, y, **fit_params):
         assert_true(len(X) == len(y))
         if self.check_X is not None:
             assert_true(self.check_X(X))
         if self.check_y is not None:
             assert_true(self.check_y(y))
         self.classes_ = np.unique(check_array(y, ensure_2d=False,
                                               allow_nd=True))
+        if self.expected_fit_params:
+            missing = (set(np.atleast_1d(self.expected_fit_params)) -
```

Comment:
The `np.atleast_1d` call converts a string to a one-element array, so that users could give a string input to `expected_fit_params` instead of a one-element list.

In this case, without the array casting there would be an obvious and loud error if someone forgot to make that input a list, and there's not much benefit to saving two characters for a test helper. I'll remove the `np.atleast_1d`.